components/custom_status/custom_status_modal.tsx

```diff
@@ -170,39 +178,42 @@ const CustomStatusModal: React.FC<Props> = (props: Props) => {
         </div>
     );
 
+    const renderCustomStatusSuggestions = () => {
+        const recentCustomStatusTexts = recentCustomStatuses.map((status: any) => status.text);
+        const customStatusSuggestions: any[] = [];
+        suggestedCustomStatuses.map((status: UserCustomStatus) => {
+            if (!recentCustomStatusTexts.includes(status.text)) {
+                customStatusSuggestions.push(
+                    <CustomStatusSuggestion
+                        handleSuggestionClick={handleSuggestionClick}
+                        emoji={status.emoji}
+                        text={status.text}
+                    />,
+                );
+            }
+            return null;
+        });
```

Comment:
```suggestion
        const customStatusSuggestions = defaultCustomStatusSuggestions
            .filter((status: UserCustomStatus) => !recentCustomStatusTexts.includes(status.text))
            .map((status: UserCustomStatus) => (
                    <CustomStatusSuggestion
                        handleSuggestionClick={handleSuggestionClick}
                        emoji={status.emoji}
                        text={status.text}
                    />,
                ));
```

Reply:
In your suggested approach we are iterating the array twice... isn't is better if we iterate only once?

Reply:
1. Some browsers will optimize the chaining of operators.
2. Iterating an array of length 5 once or twice is not that much optimization.
3. The code is much more readable this way.